### PR TITLE
feat(react-wallet): Create app layout with left nav and app bar

### PR DESCRIPTION
**Pull request by samsiegart**
## Description

This creates the basic app layout with an app bar, left nav, and routing between views. It uses `material-ui` for components/styling and `react-router-dom` for routing. Some TODOs are making the wallet connection status into a proper button/menu, and making the left nav responsive with a collapsible drawer on narrow viewports.

Screenshot: 
image

### CODE REVIEW

File: packages/dapp-svelte-wallet/react-ui/src/App.js [hunk 1]
```diff
@@ -1,22 +1,85 @@
...
-        Connection Status: {connectionState}
-      </header>
-      <WalletConnection></WalletConnection>
-    </div>
+    <ThemeProvider theme={appTheme}>
+      <CssBaseline />
+      <NavMenu />
+      <main className={classes.main}>
+        <Switch>
+          <Route path="/purses">
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/react-ui/src/App.js:63]: Are these Route paths compatible with installing the wallet under `http://localhost:8000/wallet/`?

File: packages/dapp-svelte-wallet/react-ui/src/components/NavMenu.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  nav: {
+    position: 'fixed',
+    top: theme.appBarHeight,
+    width: theme.navMenuWidth,
+    height: `calc(100vh - ${theme.appBarHeight})`,
+    overflowY: 'auto',
+  },
+  sectionHeader: {
+    padding: '16px',
+    fontFamily: ['Montserrat', 'Arial', 'sans-serif'].join(','),
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/react-ui/src/components/NavMenu.js:23]: Omitting the double-quotes from `'"Montserrat"'`, etc seems inconsistent with the other `fontFamily` style.

File: packages/dapp-svelte-wallet/react-ui/src/App.js [hunk 3]
```diff
@@ -1,22 +1,85 @@
...
 
-const App = ({ connectionState }) => {
+const appTheme = createTheme({
+  palette: {
+    background: {
+      default: '#ffffff',
+    },
+  },
+  typography: {
+    fontFamily: ['"Roboto"', '"Helvetica"', '"Arial"', 'sans-serif'].join(','),
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/react-ui/src/App.js]: Using literal double-quotes here seems inconsistent with other usage.  I could see that `'"Roboto","Helvetica","Arial",sans-serif'` is a legitimate style value, but it seems strange to add the quotes if they are not needed.

File: packages/dapp-svelte-wallet/react-ui/src/components/WalletConnection.js [hunk 4]
```diff
@@ -78,12 +94,20 @@ const WalletConnection = ({ setConnectionState }) => {
   }, []);
 
   return (
-    <span className="hidden">
-      <AgoricWalletConnection onState={onWalletState} />
-    </span>
+    <div className={classes.connector}>
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/react-ui/src/components/WalletConnection.js:97]: Great, this is exactly how I intended the wallet connection component to be used!  Hidden, with some other component exposing state.

File: packages/dapp-svelte-wallet/react-ui/src/tests/App.test.js [hunk 5]
```diff
@@ -1,7 +1,18 @@
...
+import { MemoryRouter } from 'react-router-dom';
 import App from '../App';
+import Dashboard from '../views/Dashboard';
+import Purses from '../views/Purses';
+import Dapps from '../views/Dapps';
+import Contacts from '../views/Contacts';
+import Issuers from '../views/Issuers';
 
 jest.mock('../components/WalletConnection', () => () => 'WalletConnection');
+jest.mock('../views/Dashboard', () => () => 'Dashboard');
```
Comment (michaelfig) [inline packages/dapp-svelte-wallet/react-ui/src/tests/App.test.js:11]: Nice!

**Review by michaelfig — APPROVED** [review]
LGTM.  Good step forward!

File: packages/dapp-svelte-wallet/react-ui/src/App.js (hunk 3 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/react-ui/src/App.js]: Fixed to just use single quotes, nice catch (it was just copypasta)

File: packages/dapp-svelte-wallet/react-ui/src/components/NavMenu.js (hunk 2 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/react-ui/src/components/NavMenu.js:23]: Fixed in the other place

File: packages/dapp-svelte-wallet/react-ui/src/tests/App.test.js (hunk 5 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/react-ui/src/tests/App.test.js:11]: 💯

File: packages/dapp-svelte-wallet/react-ui/src/components/WalletConnection.js (hunk 4 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/react-ui/src/components/WalletConnection.js:97]: Yea I think this works pretty well!

File: packages/dapp-svelte-wallet/react-ui/src/App.js (hunk 1 above)
Comment (samsiegart) [reply packages/dapp-svelte-wallet/react-ui/src/App.js:63]: No I don't believe so. We can specify the basename "/wallet" as documented here https://reactrouter.com/web/api/BrowserRouter/basename-string to make that work. However, I'll hold off on that until we make the react wallet the default. We probably want to do it dynamically depending on how the wallet is being loaded (local dev vs agoric open).

**Pull request merged**